src/boot/boot.go

```diff
@@ -47,6 +47,7 @@ func Run() {
 	logrus.Infof("ssaplayground: welcome to ssaplayground service... http://%s/gossa", config.Get().Addr)
 	err := server.ListenAndServe()
 	if err != http.ErrServerClosed {
+		terminated <- true
```

Comment:
Why this is necessary and how will it not conflict with the code at line 44?

Reply:
I'm sorry to reply to you for such a long time. I don't know why domestic GitHub alwayse can't get on. I have the following reasons for the modification:

1. When most programs start, if there is a port conflict, the program will exit, so I consider whether I can add this function;

2. After adding the "terminated < - true", there will be no conflict with line 44, because if there is a port conflict, the HTTP server will directly return error at startup and put the data in "terminated". Then the main protocol will take out the data in "terminated channel" and exit. After verification, there is no problem. I don't know whether it is appropriate to do so.

Reply:
Sorry for the very late response. This is a valid request. Thanks a lot.